### DISCUSSION TITLE
Define which YAML file storing variables to use in _quarto.yml

### DISCUSSION THREAD

**Discussion by ToryDeng** [quarto-dev/quarto-cli#7367] [Q&A] [answered]
### Description

Is it possible to specify which YAML file to use in the `_quarto.yml`, allowing me to switch between different sets of variables? After going through the documentation, it appears that I can only create one YAML file, namely `_variables.yml`, to store variables. However, I want to use multiple YAML files to keep things organized and clear. For example, I'd like to create 2 YAML files, each for a laboratory, like `_variables_Lab_A.yml` and `_variables_Lab_B.yml`:
```yml
# _variables_Lab_A.yml
address: 
email: 

# _variables_Lab_B.yml
address: 
email: 
```
and the `_quarto.yml`:

```yml
variable: 
  default: Lab_A
```

Then I can write a sentence like this in qmd files: `Our email is {{< var email >}}`. When rendering the doc for Lab_B, I just need to change the `_quarto.yml` without modifying the content in qmd files. Otherwise if I put all variables in the `_variables.yml`:
```yml
# _variables.yml
Lab_A:
  address:
  email:
Lab_B:
  address:
  email:
```

The content will be `Our email is {{< var Lab_A.email >}}` and I have to modify it when I want to render doc for Lab_B. 

A workaround is to use two profiles `_quarto-Lab_A.yml` and `_quarto-Lab_B.yml`, and use the `when-profile` option to switch between variable sets:

```qmd
::: {.content-visible when-profile="Lab_A"}
Our email is {{< var Lab_A.email >}}.
:::

::: {.content-visible when-profile="Lab_B"}
Our email is {{< var Lab_B.email >}}.
:::
```

The `_quarto.yml` is
```yml
profile: 
  default: Lab_A
```
But this is a little redundant. Maybe I can define variables in the two profiles but I think variables should be in `_variable_*.yml`, not metadata files (sorry I act as if I have a bit of OCD, but this is a part of who I am :joy:).

**Comment by mcanouil** [comment, marked as answer]
`_variables.yml` is not conditional on profiles.
`_variables.yml` is not the same thing as `_quarto.yml`, *i.e.*, the variable shortcode don't access `_quarto.yml` but the meta shortcode does. `_quarto.yml` stores metadata.

If you want variables to depend on profiles, then the conditional content approach is the good and only one for now at least.

See:
- https://github.com/quarto-dev/quarto-cli/issues/5954

**Reply by ToryDeng** [reply]
Wow, it's great to hear that you've already been working on this enhancement. Many thanks, and I'm looking forward to it!
